app/scripts/modules/core/src/presentation/modal/TabBoundary.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { useEventListener } from 'core';
+
+export const TabBoundary = ({ children }: { children: React.ReactNode }) => {
+  const dialogRef = React.createRef<HTMLDivElement>();
```

Comment:
Can we try moving this whole component up to a top level `presentation` component and calling this variable something other than `dialogRef`? Or is there something about it that makes it modal-specific?

Reply:
I don't know if there are other use cases outside of modals, but there's nothing really about the component that's specific to modal, so it makes sense to move it up to `presentation`.

The `dialogRef` variable name is because I did a lazy refactor. I initially wrote this within the `Modal` component, then copy/pasted it into its own component and didn't look at the code before clicking the button to start the code review. I will rename it to something appropriate.